core/server/worker/src/main/java/alluxio/worker/AlluxioWorkerProcess.java

```diff
@@ -56,125 +56,143 @@ public final class AlluxioWorkerProcess implements WorkerProcess {
-      }
-      // Setup GRPC server
-      mDataServer = new GrpcDataServer(mRpcConnectAddress.getHostName(), mRpcBindAddress, this);
 
       // Setup Netty Data Server
       mNettyDataTransmissionEnable =
           Configuration.global().getBoolean(PropertyKey.USER_NETTY_DATA_TRANSMISSION_ENABLED);
       if (mNettyDataTransmissionEnable) {
-        mNettyDataServerAddress =
-            NetworkAddressUtils.getBindAddress(ServiceType.WORKER_DATA, Configuration.global());
-        mNettyDataServer = new NettyDataServer(mNettyDataServerAddress, this);
+        mNettyDataServer = nettyDataServer;
       } else {
-        mNettyDataServerAddress = null;
         mNettyDataServer = null;
       }
-
-      // Setup domain socket data server
-      if (isDomainSocketEnabled()) {
-        String domainSocketPath =
-            Configuration.getString(PropertyKey.WORKER_DATA_SERVER_DOMAIN_SOCKET_ADDRESS);
-        if (Configuration.getBoolean(PropertyKey.WORKER_DATA_SERVER_DOMAIN_SOCKET_AS_UUID)) {
-          domainSocketPath =
-              PathUtils.concatPath(domainSocketPath, UUID.randomUUID().toString());
-        }
-        LOG.info("Domain socket data server is enabled at {}.", domainSocketPath);
-        mDomainSocketDataServer = new GrpcDataServer(mRpcConnectAddress.getHostName(),
-            new DomainSocketAddress(domainSocketPath), this);
-        // Share domain socket so that clients can access it.
-        FileUtils.changeLocalFileToFullPermission(domainSocketPath);
-      }
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Set up servers with block worker.
+   * @param dataServerFactory
+   */
+  private void setUpServersWithBlockWorker(DataServerFactory dataServerFactory) {
+    // Setup web server
+    mWebServer =
+        new WorkerWebServer(NetworkAddressUtils.getBindAddress(ServiceType.WORKER_WEB,
+            Configuration.global()), this,
+            mRegistry.get(BlockWorker.class));
+    // Setup GRPC server
+    mDataServer = dataServerFactory.createRemoteGrpcDataServer(
+        mRegistry.get(BlockWorker.class));
+    // Setup domain socket data server
+    if (isDomainSocketEnabled()) {
+      mDomainSocketDataServer = dataServerFactory.createDomainSocketDataServer(
+          mRegistry.get(BlockWorker.class));
+    }
+  }
+
+  private void setUpServersWithDoraWorker(DataServerFactory dataServerFactory) {
+    // Setup web server
+    mWebServer =
+        new WorkerWebServer(NetworkAddressUtils.getBindAddress(ServiceType.WORKER_WEB,
+            Configuration.global()), this,
+            mRegistry.get(DoraWorker.class));
+    // Setup GRPC server
+    mDataServer = dataServerFactory.createRemoteDoraGrpcDataServer(
+        mRegistry.get(DoraWorker.class));
+    // Setup domain socket data server
+    if (isDomainSocketEnabled()) {
+      mDomainSocketDataServer = dataServerFactory.createDomainSocketDoraDataServer(
+          mRegistry.get(DoraWorker.class));
+    }
+  }
```

Comment:
Can we define a single function which takes two parameters?
```
 private void setUpServersWithWorker(DataServerFactory dataServerFactory, Class<DataWorker> c) {
     // Setup web server
     mWebServer =
         new WorkerWebServer(NetworkAddressUtils.getBindAddress(ServiceType.WORKER_WEB,
             Configuration.global()), this,
             mRegistry.get(c));
     // Setup GRPC server
     mDataServer = dataServerFactory.createRemoteDoraGrpcDataServer(
         mRegistry.get(c));
     // Setup domain socket data server
     if (isDomainSocketEnabled()) {
       mDomainSocketDataServer = dataServerFactory.createDomainSocketDoraDataServer(
           mRegistry.get(c));
     }
   }

```

and call them 
setUpServersWithWorker(dataServerFactory, DoraWorker.class)
and 
setUpServersWithWorker(dataServerFactory, BlockWorker.class)

Reply:
We can combine these two methods into one, because `mRegistry.get(Class<W> clazz)` cannot get any object if we give it a `DataWorker.class` type. Although both `BlockWorker` and `DoraWorker` extend `DataWorker`,  we can only get either `BlockWorker.class` or `DoraWorker.class`, but not `DataWorker.class`.